### ISSUE TITLE
A BUG in PRM dataset placeholder tokenizer processing

### ISSUE THREAD

**Issue by mzf666** [OpenRLHF/OpenRLHF#683]
Dear OpenRLHF contributors,

Recently, I was trying to train a PRM model using the script in `openrlhf/examples/scripts/train_prm_mistral.sh`. 
```{python}
read -r -d '' training_commands <<EOF
openrlhf.cli.train_prm \
   --save_path ./checkpoint/mistal-7b-prm \
   --save_steps 500 \
   --logging_steps 1 \
   --eval_steps 100 \
   --train_batch_size 256 \
   --micro_train_batch_size 8 \
   --pretrain mistralai/Mistral-7B-v0.1  \
   --bf16 \
   --max_epochs 1 \
   --max_len 8192 \
   --zero_stage 3 \
   --learning_rate 1e-6 \
   --dataset zhuzilin/Math-Shepherd \
   --input_key input \
   --label_key value \
   --flash_attn \
   --load_checkpoint \
   --gradient_checkpointing \
   --packing_samples \
   --wandb_group prm \
   --placeholder_token ки \
   --reward_tokens + -
EOF
     # --use_wandb [WANDB_TOKENS] or True (use wandb login command)
     # --packing_samples

if [[ ${1} != "slurm" ]]; then
    deepspeed --include localhost:0,3 --module $training_commands
fi

```

However, when I dive into the `__getitem__` function in `ProcessRewardDataset` class by running `x = train_dataset[0]` at line 61 of `openrlhf/cli/train_prm.py` as follows, it seems that the `__getitem__` function does not work as expected.

Specifically, all the `labels` equals to `-100`. I dived into the `__getitem__` function, it seems that this is because when setting `place_holder_token = 'ки'`, the tokenizer encode `'ки'` as `17165` (by running `self.tokenizer.encode(self.placeholder_token, add_special_tokens=False)`). 

However, when `'ки'` appears in the sentence, the token id of `'ки'` changes:
```{python}
text = 'Hello every one ки\nI am good. ки'
print(self.tokenizer(text, add_special_tokens=False))
"""
{'input_ids': [9906, 1475, 832, 116624, 198, 40, 1097, 1695, 13, 116624], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
"""
```
In this case, I can no longer trace the occurrence of `'ки'` according to the token-id `17165`.

This contradict to the preprocessing step in `ProcessRewardDataset.__getitem__`,  as it traces the location of the placeholder by the highlighted row (according to the `self.tokenizer.encode('ки', add_special_tokens=False)` which equals `17165 `)

![Image](https://github.com/user-attachments/assets/64468c4f-5289-42d8-b34a-1b165b2a4000)

Is there any solution regarding this?

**Comment by mzf666** [comment]
It seems that `' ки'` will be considered as a whole by the tokenizer. Does this leads to systematic error in PRM workflows?

![Image](https://github.com/user-attachments/assets/a0662839-ad8d-41e4-ab20-1ba7e24cb7a6)

**Comment by mzf666** [comment]
It seems that, we can set `--placeholder_token` as `' ки'` rather than `'ки'` to avoid the labels to be fully `[-100,-100,...,-100]`. I wonder whether this fix can solve this problem fundamentally.

**Comment by LinXueyuanStdio** [comment]
Issue #645 PR: #648 

You should choose a placeholder_token based on your base model. Then replace `'ки'` with your chosen placeholder_token to create a new training set that only for your base model.
